### PR TITLE
NET-260 REST API tests 

### PR DESCRIPTION
**Pull request by TobiaszCudnik**
## Describe your changes

NET-260 REST API tests for `GET /api/v1/enrollment-keys`.

## Provide testing steps

`go test ./... -v -tags integration -test.run TestHasNetworksAccessAPI`

## Checklist before requesting a review
- [ ] My changes affect only 10 files or less.
- [ ] I have performed a self-review of my code and tested it.
- [ ] If it is a new feature, I have added thorough tests, my code is <= 1450 lines.
- [ ] If it is a bugfix, my code is <= 200 lines.
- [ ] My functions are <= 80 lines.
- [ ] I have had my code reviewed by a peer.
- [ ] My unit tests pass locally.
- [ ] Netmaker is awesome.

### CODE REVIEW

**Review by mattkasun — CHANGES_REQUESTED** [review]
fix order of imports in database/sqlite.go

File: database/sqlite.go [hunk 1]
```diff
@@ -3,14 +3,23 @@ package database
 import (
 	"database/sql"
 	"errors"
+	_ "github.com/mattn/go-sqlite3" // need to blank import this package
```
[the remark this answers is not in the document]
Comment (TobiaszCudnik) [reply database/sqlite.go]: Fixed.

`goimports` should be added to precommit. `gofmt` [orders it differently](https://github.com/golang/go/issues/37719).

**Review by Aceix — APPROVED** [review]

**Review by theguy951357 — APPROVED** [review]
kitemake ticket is listed in on hold column so I didn't move it, but pr steps pass and critical list passes.

File: test/enrollmentkey_test.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+	}
+	adminConfig := userConfig
+	adminConfig.MasterKey = "foo123"
+	adminConfigBad := userConfig
+	adminConfigBad.MasterKey = "wrongpass"
+	adminConfigBad.Password = "wrongpass"
+	config.SetContext("user-ctx-1", userConfig)
+	config.SetContext("admin-ctx-1", adminConfig)
+	config.SetContext("admin-ctx-2", adminConfigBad)
+	config.SetCurrentContext("user-ctx-1")
```
Comment (abhishek9686) [inline test/enrollmentkey_test.go:71]: what is the role of `SetContext` and `SetCurrentContext` here?

**Review by abhishek9686 — CHANGES_REQUESTED** [review]

File: test/enrollmentkey_test.go (hunk 2 above)
Comment (TobiaszCudnik) [reply test/enrollmentkey_test.go:74]: It sets the configs (as in `nmctl context list`). Ive added some comments to those. Those arent golang's context.

Comment (abhishek9686) [reply test/enrollmentkey_test.go:74]: But isn't these tests for apis?
why is context set here for cli?

Comment (TobiaszCudnik) [reply test/enrollmentkey_test.go:74]: It uses the httpclient from `netmaker/cli/functions` (via `GetEnrollmentKeys` in this case), which is a good thing, although that httpclient is too limited ATM.

File: test/enrollmentkey_test.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+	go controller.HandleRESTRequests(&wg, ctx)
+	// TODO make sure that HTTP is up
+	time.Sleep(1 * time.Second)
+
+	t.Run("normal user", func(t *testing.T) {
+		keys := *functions.GetEnrollmentKeys()
+		assert.Len(t, keys, 1, "1 key expected")
+		assert.Len(t, keys[0].Networks, 1, "Key with 1 network expected")
+		assert.Equal(t, keys[0].Networks[0], n1.NetID, "Network ID matches")
```
Comment (gabrielseibel1) [inline test/enrollmentkey_test.go:122]: are these messages not displayed on failure? honestly thought that but am not sure

File: test/enrollmentkey_test.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+	})
+
+	// TODO assert no access
+	t.Run("incorrect masteradmin", func(t *testing.T) {
+		t.Skip("Skipping until err exposed")
+		config.SetCurrentContext("admin-ctx-2")
+		// TODO doesnt return err
+		res := *functions.GetEnrollmentKeys()
+		fmt.Println(res)
+		//assert.Error(t, res, "403 error")
```
Comment (gabrielseibel1) [inline test/enrollmentkey_test.go:139]: should probably have an assertion here, len 0?

**Review by gabrielseibel1 — CHANGES_REQUESTED** [review]
assertion for last test?

**Pull request closed without merging**